pxr/imaging/plugin/hdRpr/mesh.cpp

```diff
@@ -90,33 +90,94 @@ void HdRprMesh::Sync(
 	{
 		HdMeshTopology meshTopology = GetMeshTopology(sceneDelegate);
 
+		const VtIntArray & indexes = meshTopology.GetFaceVertexIndices();
+		const VtIntArray & vertexPerFace = meshTopology.GetFaceVertexCounts();
+
+		HdPrimvarDescriptorVector primvarDescsPerInterpolation[HdInterpolationCount];
+		for (int i = 0; i < HdInterpolationCount; ++i) {
+			auto interpolation = static_cast<HdInterpolation>(i);
+			primvarDescsPerInterpolation[i] = sceneDelegate->GetPrimvarDescriptors(id, interpolation);
+		}
+		auto getPrimvarInterpolation = [&primvarDescsPerInterpolation](pxr::TfToken const& token) -> HdInterpolation {
+			auto it = std::find_if(primvarDescsPerInterpolation, primvarDescsPerInterpolation + HdInterpolationCount,
+				[&token](HdPrimvarDescriptorVector const& descriptors) {
+				return std::any_of(descriptors.begin(), descriptors.end(), [&token](HdPrimvarDescriptor const& descriptor) {
+					return descriptor.name == token;
+				});
+			});
+			return static_cast<HdInterpolation>(std::distance(primvarDescsPerInterpolation, it));
+		};
+
 		VtValue value;
 		value = sceneDelegate->Get(id, HdTokens->points);
 		VtVec3fArray points = value.Get<VtVec3fArray>();
 
 		VtVec2fArray st;
-
-
-		value = sceneDelegate->Get(id, TfToken("st"));
+		float timeSamples[1] = {0.0f};
```

Comment:
What happens if ST is varying over time, does this always get the first timesample?  Or just the one at current time?

Reply:
This gets first or default timesample. Unfortunately, there is no way to get the current time from HdSceneDelegate, only UsdImagingDelegate knows it.

Let's see what USD team thinks about adding a virtual method GetTime to HdSceneDelegate. [Issue](https://github.com/PixarAnimationStudios/USD/issues/897)

Reply:
I totally missed that SamplePrimvar accepts not absolute time but relative to the current time.
So present changes do fully handle varying over time ST